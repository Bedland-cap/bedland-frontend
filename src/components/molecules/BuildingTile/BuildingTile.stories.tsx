import type { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';

import BuildingTile from './BuildingTile';

const meta = {
  title: 'Basic Components/Molecules/BuildingTile',
  component: BuildingTile,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof BuildingTile>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Main: Story = {
  args: {
    BuildingAddress: 'Berry Street 27',
    BuildingId: 4293,
    BuildingImg: 'assets/BuildingImg.png',
  },
};
