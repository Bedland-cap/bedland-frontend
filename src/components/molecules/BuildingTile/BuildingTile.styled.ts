import styled from 'styled-components';
import { HexColor } from 'theme/theme.types';

export const TileWrapper = styled.div<{ shadowColor: HexColor }>`
  background-color: #fff;
  border-radius: 8px;
  box-shadow: ${({ shadowColor }) => `0px 5px 8px ${shadowColor}20`};
  color: #4b4b4b;
  display: flex;
  flex-direction: column;
`;

export const BuildingImage = styled.img`
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
`;

export const CardContent = styled.div`
  padding: 0.5rem 1.031rem 0.938rem;
`;
