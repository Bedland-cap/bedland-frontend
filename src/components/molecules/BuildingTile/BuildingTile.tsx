import Typography from 'components/atoms/Typography/Typography';
import { useContext } from 'react';
import { ThemeContext } from 'theme/ThemeContext';
import BuildingImg from 'assets/img/BuildingImg.png';
import * as Styled from './BuildingTile.styled';

export type BuildingTileProps = {
  buildingId: string;
  buildingAddress: string;
  buildingImg?: string;
};

const defaultProps: { buildingImg: string } = {
  buildingImg: BuildingImg,
};

const BuildingTile = ({ buildingId, buildingAddress, buildingImg }: BuildingTileProps) => {
  const { palette } = useContext(ThemeContext);

  return (
    <Styled.TileWrapper shadowColor={palette.shadow}>
      <Styled.BuildingImage src={buildingImg} alt='Building' />
      <Styled.CardContent>
        <Typography variant='cardTitle' color='inputGrey'>
          Building #{buildingId}
        </Typography>
        <Typography variant='paragraph' color='inputGrey'>
          {buildingAddress}
        </Typography>
        <Styled.ButtonWrapper>
          <Styled.Button palette={palette}>Open details</Styled.Button>
        </Styled.ButtonWrapper>
      </Styled.CardContent>
    </Styled.TileWrapper>
  );
};

BuildingTile.defaultProps = defaultProps;

export default BuildingTile;
