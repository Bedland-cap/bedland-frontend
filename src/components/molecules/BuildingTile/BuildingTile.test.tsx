import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from 'theme/ThemeContext';

import BuildingTile from './BuildingTile';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider>
        <BuildingTile
          BuildingId={4293}
          BuildingImg='/assets/BuildingImg.png'
          BuildingAddress='Berry Street 27'
        />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
