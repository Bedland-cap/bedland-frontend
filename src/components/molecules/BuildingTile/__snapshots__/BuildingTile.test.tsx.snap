// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bcXHqe dJOXqN"
>
  <img
    alt="Building"
    className="sc-gswNZR djSoAM"
    src="/assets/BuildingImg.png"
  />
  <h2
    className="sc-dkrFOg izGeNl"
  >
    Building #
    4293
  </h2>
  <p
    className="sc-hLBbgP iunxbB"
  >
    Berry Street 27
  </p>
  <div
    className="sc-eDvSVe kIwdaK"
  >
    <button
      className="sc-jSUZER jCSFFe"
    >
      Open details
    </button>
  </div>
</div>
`;
